### PR TITLE
[Fixes #8422] Added pluggable service types

### LINKED ISSUE #8422: Add a hook for external service types implementation
**Is your feature request related to a problem? Please describe.**
We need to add new services type without modify the core each time.

**Describe the solution you'd like**
We implemented a code similar to metadata parser and stores, where the new type can be ingested throught a list of modules which have the variable _services_type_

### PR DESCRIPTION
**Pull request by mattiagiupponi**
- Service type are now pluggable like we did in past with [metadata_storer](https://docs.geonode.org/en/master/basic/settings/index.html?#metadata-storers) and [metadata_parser](https://docs.geonode.org/en/master/basic/settings/index.html?#metadata-parsers) through `SERVICES_TYPE_MODULES` variable
- The list of services available is now available through `get_available_service_type` function (will return default + custom from external libs)
- Fix choices from form to use `get_available_service_type` function
- tests coverage

Refs [#8422](https://github.com/GeoNode/geonode/issues/8422)
Doc: https://github.com/GeoNode/documentation/pull/204

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [x] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [x] The issue connected to the PR must have Labels and Milestone assigned
- [ ] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [x] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the QA checks: flake8 geonode
- [x] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates
- [ ] Commits adding **new texts** do use gettext and have updated .po / .mo files (without location infos)

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

**Comment by afabiani** [comment]
@mattiagiupponi LGTM raised some new alerts, can you please take a look?

File: geonode/services/serviceprocessors/handler.py [hunk 1]
```diff
@@ -21,30 +21,18 @@
...
-        # enumerations.TMS: {"OWS": False, "handler": ServiceHandlerBase},
-        enumerations.REST_MAP: {"OWS": False, "handler": ArcMapServiceHandler},
-        enumerations.REST_IMG: {"OWS": False, "handler": ArcImageServiceHandler},
-        # enumerations.CSW: {"OWS": False, "handler": ServiceHandlerBase},
-        # enumerations.HGL: {"OWS": True, "handler": ServiceHandlerBase},  # TODO: verify this
-        # enumerations.OGP: {"OWS": False, "handler": ServiceHandlerBase},  # TODO: verify this
-    })
+    handlers = get_available_service_type()
+
     if service_type in (enumerations.AUTO, enumerations.OWS):
```
Comment (afabiani) [inline geonode/services/serviceprocessors/handler.py:36]: Let's get rid of this "if" statement and of the `service_type` parameter which seems to be harmless and unuseful.

**Review by afabiani — CHANGES_REQUESTED** [review]
See my comments inline other than please check the LGTM issues,

File: geonode/services/serviceprocessors/handler.py (hunk 1 above)
Comment (mattiagiupponi) [reply geonode/services/serviceprocessors/handler.py:36]: if removed, the service type is used to get the handler from the list

**Review by afabiani — APPROVED** [review]

**Pull request merged**